Python/sysmodule.c

```diff
@@ -3479,6 +3479,66 @@ make_impl_info(PyObject *version_info)
     return NULL;
 }
 
+static PyObject*
+make_abi_features(void)
+{
+    int res;
+    PyObject *features = PyList_New(0);
+    if (features == NULL) {
+        goto error;
+    }
+
+#if defined(Py_GIL_DISABLED)
+    PyObject *threading_feature = PyUnicode_FromString("free-threading");
+#else
+    PyObject *threading_feature = PyUnicode_FromString("gil-enabled");
+#endif
+    if (threading_feature == NULL) {
+        goto error;
+    }
+    res = PyList_Append(features, threading_feature);
+    Py_DECREF(threading_feature);
+    if (res < 0) {
+        goto error;
+    }
+
+#if defined(Py_DEBUG) || (defined(MS_WINDOWS) && defined(_DEBUG))
+    PyObject *debug = PyUnicode_FromString("debug");
+    if (debug == NULL) {
+        goto error;
+    }
+    res = PyList_Append(features, debug);
+    Py_DECREF(debug);
+    if (res < 0) {
+        goto error;
+    }
+#endif
+
+    bool is64bit = PY_SSIZE_T_MAX > (1L << 32);
```

Comment:
This is true for 32-bit systems as well, because 1 << 32 will overflow. Maybe something like `UINTPTR_MAX == 0xffffffff` is better? I'm not sure about it though.